### ISSUE TITLE
snapshot testing

### ISSUE THREAD

**Issue by bahmutov** [update-readme/update-readme#1]
For testing, a good idea is to use either Ava's built-in snapshot testing or something like my own https://github.com/bahmutov/snap-shot or https://github.com/bahmutov/schema-shot

**Comment by billiegoose** [comment]
Yes! That would be great. I hadn't added any yet because I am still very open to changing the parsed representation, but I *definitely* want to upgrade the tests from "read a handful of files synchronously in a for loop" to something more structured. Also, the current fixtures were generated by [lorem markdownum](https://jaspervdj.be/lorem-markdownum/) and are not representative of actual READMEs.

I really enjoyed writing `test/all-the-readmes.js` and while running it on over 400,000 READMEs has done a great job helping me weed out obscure corner cases (who puts a carriage return with no newline in their readme?) it took well over an hour to run. It would be awesome to have a way to test on a "representative sample" and a test suite that can't be thwarted by network failure. (I tried to achieve the latter, but the offline version using `all-the-packages` instead of `package-stream` [was missing](https://github.com/nice-registry/all-the-packages/issues/3) an enormous number of readmes.) If we had that, we could put the test suite in Travis and enjoy the happy green Test Passed messages that Github sprinkles wherever commits are found.
